torch/_export/__init__.py

```diff
@@ -352,6 +382,26 @@ def export(
     )
     gm_torch_level.recompile()
 
+    param_buffer_table = {}
+    if isinstance(f, torch.nn.Module):
+        param_lookup = {}
+        buffer_lookup = {}
+        for name, param in f.named_parameters():
+            param_lookup[id(param)] = name
+        for name, buffer in f.named_buffers():
+            buffer_lookup[id(buffer)] = name
+        for dynamo_name, dynamo_param in gm_torch_level.named_parameters():
+            assert dynamo_name not in param_buffer_table
+            if id(dynamo_param) in param_lookup:
+                param_buffer_table[dynamo_name] = param_lookup[id(dynamo_param)]
```

Comment:
Are you mapping dynamo param name to the original name?

Reply:
yes